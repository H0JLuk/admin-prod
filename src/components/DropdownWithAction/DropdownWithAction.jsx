import React, { useCallback, useMemo, useState } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { Dropdown, Button } from 'antd';
import { DownOutlined, UpOutlined } from '@ant-design/icons';

import styles from './DropdownWithAction.module.css';

const RESET_LABEL = 'Сбросить';

const DROPDOWN_ARROW_STYLE = {
    fontSize: '14px',
    color: '#c3c9d0',
};

const DropdownWithAction = ({
    dropdownLabel,
    onMenuItemClick,
    menuItems = [],
    withReset,
}) => {
    const [dropdownActive, setDropdownActive] = useState(false);

    const menuItemClickHandler = useCallback((name) => {
        setDropdownActive(false);
        onMenuItemClick(name);
    }, [onMenuItemClick]);

    const overlay = useMemo(() => (
        <div className={ styles.dropdownMenu }>
            {menuItems.map((item) => (
                <div
                    key={ item.name }
                    className={ cn(styles.dropdownMenuItem, { [styles.active]: item.active }) }
                    onClick={ () => menuItemClickHandler(item.name) }
                >
                    { item.label }
                </div>
            ))}
            {withReset && (
                <div
                    className={ styles.dropdownMenuItem }
                    onClick={ menuItemClickHandler }
                >
                    { RESET_LABEL }
                </div>
            )}
        </div>
    ), [menuItems, menuItemClickHandler, withReset]);

    return (
        <Dropdown
            className
            overlay={ overlay }
            trigger={ ['click'] }
            visible={ dropdownActive }
            onVisibleChange={ setDropdownActive }
        >
            <Button
                className={ cn(styles.dropdownButton, { [styles.dropdownActive]: dropdownActive }) }
            >
                { dropdownLabel } {
                    !dropdownActive
                        ? <DownOutlined style={ DROPDOWN_ARROW_STYLE } />
                        : <UpOutlined style={ DROPDOWN_ARROW_STYLE } />
                }
            </Button>
        </Dropdown>
    );
};

DropdownWithAction.propTypes = {
    dropdownLabel: PropTypes.string.isRequired,
    onMenuItemClick: PropTypes.func.isRequired,
    menuItems: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        active: PropTypes.bool,
    })).isRequired,
    withReset: PropTypes.bool,
};

export default DropdownWithAction;
