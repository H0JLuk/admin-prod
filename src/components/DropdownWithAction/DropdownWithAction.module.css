.dropdownMenu {
    min-width: 250px;
    padding: 5px 0px;
    background-color: #ffffff;
    box-shadow: 0px 0px 10px 0px #cccccc;
}

button.dropdownButton {
    display: flex;
    align-items: center;
}

button.dropdownButton svg {
    transition: transform 0.3s;
}

button.dropdownButton:global(.ant-dropdown-open) svg {
    transform: rotate(-180deg);
}

.dropdownMenuItem {
    padding: 10px;
    cursor: pointer;
    transition: color 0.2s;
}

.dropdownMenuItem:hover {
    color: #2ac673;
}

.dropdownMenuItem.active {
    color: #09a552;
}
