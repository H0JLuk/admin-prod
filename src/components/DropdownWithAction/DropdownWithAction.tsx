import React, { memo, useCallback, useMemo, useState, MouseEvent } from 'react';
// import PropTypes from 'prop-types';
import cn from 'classnames';
import { Dropdown, Button } from 'antd';
import { DownOutlined } from '@ant-design/icons';
import { MenuItem } from '@components/HeaderWithActions/types';

import styles from './DropdownWithAction.module.css';

const RESET_LABEL = 'Сбросить';

const DROPDOWN_ARROW_STYLE = {
    fontSize: '14px',
    color: '#c3c9d0',
};

type DropdownWithActionProps = {
    dropdownLabel: string;
    onMenuItemClick: (val: string | MouseEvent<HTMLDivElement>) => void;
    menuItems: MenuItem[];
    resetLabel?: string;
    withReset?: boolean;
};

// InferProps<DropdownWithActionProps>
const DropdownWithAction: React.FC<DropdownWithActionProps> = ({
    dropdownLabel,
    onMenuItemClick,
    menuItems,
    withReset,
    resetLabel = RESET_LABEL,
}) => {
    const [dropdownActive, setDropdownActive] = useState(false);

    const menuItemClickHandler = useCallback((name: string | MouseEvent<HTMLDivElement>) => {
        setDropdownActive(false);
        onMenuItemClick(name);
    }, [onMenuItemClick]);

    const overlay = useMemo(() => (
        <div className={styles.dropdownMenu}>
            {menuItems.map((item) => (
                <div
                    key={item.name}
                    className={cn(styles.dropdownMenuItem, { [styles.active]: item.active })}
                    onClick={() => menuItemClickHandler(item.name)}
                >
                    {item.label}
                </div>
            ))}
            {withReset && (
                <div
                    className={styles.dropdownMenuItem}
                    onClick={menuItemClickHandler}
                >
                    {resetLabel}
                </div>
            )}
        </div>
    ), [menuItems, menuItemClickHandler, withReset, resetLabel]);

    return (
        <Dropdown
            overlay={overlay}
            trigger={['click']}
            visible={dropdownActive}
            onVisibleChange={setDropdownActive}
        >
            <Button className={styles.dropdownButton}>
                {dropdownLabel} {<DownOutlined style={DROPDOWN_ARROW_STYLE} />}
            </Button>
        </Dropdown>
    );
};

// TODO: Решить проблему с типизацией menuItems
// DropdownWithAction.propTypes = {
//     dropdownLabel: PropTypes.string.isRequired,
//     onMenuItemClick: PropTypes.func.isRequired,
//     resetLabel: PropTypes.string,
//     menuItems: PropTypes.arrayOf(PropTypes.shape({
//         name: PropTypes.string.isRequired,
//         label: PropTypes.string.isRequired,
//         active: PropTypes.bool,
//     })).isRequired,
//     withReset: PropTypes.bool,
// };

export default memo(DropdownWithAction);
