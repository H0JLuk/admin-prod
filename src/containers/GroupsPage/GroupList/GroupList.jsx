import React, { useCallback, useEffect, useRef, useState } from 'react';
import cn from 'classnames';
import { useHistory } from 'react-router-dom';
import { Button, Radio } from 'antd';
import { SyncOutlined } from '@ant-design/icons';

import Header from '../../../components/Header/Header';
import HeaderWithActions from '../../../components/HeaderWithActions/HeaderWithActions';
import TableDeleteModal from '../../../components/TableDeleteModal/TableDeleteModal';
import { getCampaignGroupList, deleteCampaignGroup } from '../../../api/services/campaignGroupService';
import { GROUPS_PAGES } from '../../../constants/route';
import { EmptyGroup, GroupListByType } from './GroupListComponents';
import {
    TYPES,
    BUTTON,
    SEARCH_INPUT,
    DROPDOWN_SORT_MENU,
    GROUP_OPTIONS,
    TITLE,
    PRE_TITLE,
    RESET_LABEL,
    sortByFieldKey,
    groupTypes,
} from './groupListConstants';

import styles from './GroupList.module.css';

const GroupList = ({ matchPath }) => {
    const [loading, setLoading] = useState(true);
    const copyBundleList = useRef([]);
    const copyRelatedCampaigns = useRef([]);
    const [groupBundlesList, setBundlesList] = useState([]);
    const [groupPromoCampaignsList, setPromoCampaignsList] = useState([]);

    const checkedItems = useRef({});
    const [selectedItems, setSelectedItems] = useState(null);
    const [selectedSection, setSelectedSection] = useState(TYPES.IDEA);
    const [modalIsOpen, setModalIsOpen] = useState(false);

    const history = useHistory();

    const dataList = selectedSection === TYPES.IDEA ? groupBundlesList : groupPromoCampaignsList;
    const { current: copyDataList } = selectedSection === TYPES.IDEA ? copyBundleList : copyRelatedCampaigns;
    const setDataList = selectedSection === TYPES.IDEA ? setBundlesList : setPromoCampaignsList;

    const isSelect = selectedItems !== null;

    const toggleSelect = () => clearSelectedItems(isSelect);
    const toggleModal = () => setModalIsOpen(prev => !prev);

    const loadData = async () => {
        setLoading(true);
        try {
            const { groups = [] } = await getCampaignGroupList();
            const { bundles, relatedCampaigns } = groups.reduce((result, group) => {
                const key = group.type === TYPES.IDEA ? groupTypes.BUNDLE : groupTypes.RELATED_CAMPAIGN;
                return {
                    ...result,
                    [key]: [...result[key], { ...group, linksLength: group.links.length }],
                };
            }, { bundles: [], relatedCampaigns: [] });

            copyBundleList.current = bundles;
            copyRelatedCampaigns.current = relatedCampaigns;
            setBundlesList(bundles);
            setPromoCampaignsList(relatedCampaigns);
        } catch (err) {
            console.error(err);
        }
        setLoading(false);
    };

    useEffect(() => {
        loadData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const clearSelectedItems = (offSelect, saveSelecting) => {
        checkedItems.current = {};
        setSelectedItems(prev => offSelect || (saveSelecting && !prev) ? null : []);
    };

    const handleSelectItem = (selectItem) => {
        setSelectedItems(prev => {
            const { id: selectedId, checked } = selectItem;
            checkedItems.current[selectedId] = checked;

            if (checked) {
                return [...prev, selectItem];
            }

            return prev.filter(({ id }) => id !== selectedId);
        });
    };

    const handleSelectAllItem = () => {
        const isAllChecked = dataList.length === selectedItems.length;
        if (isAllChecked) {
            clearSelectedItems();
        } else {
            const newSelectedItems = dataList.map(({ id, name }) => {
                checkedItems.current[id] = true;
                return { id, name };
            });
            setSelectedItems(newSelectedItems);
        }
    };

    const handleChangeSection = (e) => {
        const sectionType = e.target.value;
        clearSelectedItems(true);
        setSelectedSection(sectionType);
    };

    const refreshList = () => {
        loadData();
        clearSelectedItems(true);
    };

    const onAddGroupClick = () => {
        history.push(`${matchPath}${GROUPS_PAGES.ADD_GROUP}?type=${selectedSection.toLowerCase()}`);
    };

    const buttons = [
        isSelect ? {
            type: 'primary',
            label: dataList.length && dataList.length === selectedItems.length
                ? BUTTON.CHOOSE_ALL_REMOVE
                : BUTTON.CHOOSE_ALL,
            onClick: handleSelectAllItem,
        } : {
            type: 'primary',
            label: BUTTON.ADD,
            onClick: onAddGroupClick,
        },
        {
            label: isSelect ? BUTTON.CANCEL : BUTTON.CHOOSE,
            onClick: toggleSelect,
        }
    ];

    const onChangeInput = useCallback(() => {
        clearSelectedItems(false, true);
    }, []);

    const searchAndSortParams = {
        setDataList,
        copyDataList,
        matchPath,
        inputPlaceholder: selectedSection === TYPES.IDEA ? SEARCH_INPUT.BUNDLE_SEARCH : SEARCH_INPUT.PROMO_SEARCH,
        sortByFieldKey: sortByFieldKey.NAME,
        menuItems: DROPDOWN_SORT_MENU,
        onChangeInput,
    };

    return (
        <div className={ cn(styles.container, { [styles.notEmpty]: !!copyDataList.length && !loading }) }>
            <Header buttonBack={ false } menuMode />
            <Radio.Group
                onChange={ handleChangeSection }
                value={ selectedSection }
                className={ styles.section }
                options={ GROUP_OPTIONS }
                optionType="button"
            />
            { loading ? (
                <div className={ styles.loadingContainer }>
                    <div className={ styles.loading }>
                        <SyncOutlined spin />
                    </div>
                </div>
            ) : !copyDataList.length ? (
                <EmptyGroup
                    onCreateClick={ onAddGroupClick }
                    title={
                        selectedSection === TYPES.IDEA
                            ? TITLE.EMPTY_BUNDLES
                            : TITLE.EMPTY_PROMO_CAMPAIGNS
                    }
                    preTitle={
                        selectedSection === TYPES.IDEA
                            ? PRE_TITLE.CREATE_BUNDLE
                            : PRE_TITLE.CREATE_PROMO_CAMPAIGN
                    }
                />
            ) : (
                <>
                    <HeaderWithActions
                        title={ selectedSection === TYPES.IDEA ? TITLE.BUNDLE : TITLE.PROMO_CAMPAIGNS }
                        buttons={ buttons }
                        triggerResetParams={ selectedSection }
                        resetLabel={ RESET_LABEL }
                        { ...searchAndSortParams }
                    />
                    <div className={ styles.content }>
                        <GroupListByType
                            type={ selectedSection }
                            checkedItems={ checkedItems.current }
                            dataList={ dataList }
                            matchPath={ matchPath }
                            select={ isSelect }
                            handleSelectItem={ handleSelectItem }
                        />
                    </div>
                    { isSelect && (
                        <div className={ styles.footer }>
                            <span>{ `Выбрано групп: ${selectedItems.length}` }</span>
                            <Button
                                disabled={ !selectedItems.length }
                                onClick={ toggleModal }
                                type="primary"
                                danger
                            >
                                { BUTTON.DELETE }
                            </Button>
                        </div>
                    ) }
                    <TableDeleteModal
                        modalClose={ toggleModal }
                        sourceForRemove={ selectedItems || [] }
                        listIdForRemove={ (selectedItems || []).map(item => item.id) }
                        deleteFunction={ deleteCampaignGroup }
                        refreshTable={ refreshList }
                        modalSuccessTitle={ TITLE.MODAL_SUCCESS_TITLE }
                        visible={ modalIsOpen }
                        modalTitle={ TITLE.MODAL_TITLE }
                        listNameKey={ sortByFieldKey.NAME }
                    />
                </>
            ) }
        </div>
    );
};

export default GroupList;
