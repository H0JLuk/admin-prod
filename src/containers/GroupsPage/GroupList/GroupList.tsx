import React, { useCallback, useEffect, useRef, useState } from 'react';
import cn from 'classnames';
import { useHistory } from 'react-router-dom';
import { Button, Radio, RadioChangeEvent } from 'antd';
import { SyncOutlined } from '@ant-design/icons';
import Header from '@components/Header';
import HeaderWithActions, { ButtonProps } from '@components/HeaderWithActions';
import TableDeleteModal from '@components/TableDeleteModal';
import { getCampaignGroupList, deleteCampaignGroup } from '@apiServices/campaignGroupService';
import { GROUPS_PAGES } from '@constants/route';
import { EmptyGroup, GroupListByType } from './GroupListComponents';
import {
    SEARCH_INPUT,
    DROPDOWN_SORT_MENU,
    GROUP_OPTIONS,
    TITLE,
    PRE_TITLE,
    RESET_LABEL,
    groupTypes,
} from './groupListConstants';
import { BUTTON_TEXT } from '@constants/common';
import { BundleTypes } from '../groupPageConstants';
import { BundleDtoWithLinksLength, BundlesRelatedCampaigns, CheckedItem } from './types';

import styles from './GroupList.module.css';

type GroupListProps = {
    matchPath: string;
};

const GroupList: React.FC<GroupListProps> = ({ matchPath }) => {
    const [loading, setLoading] = useState(true);
    const copyBundleList = useRef<BundleDtoWithLinksLength[]>([]);
    const copyRelatedCampaigns = useRef<BundleDtoWithLinksLength[]>([]);
    const [groupBundlesList, setBundlesList] = useState<BundleDtoWithLinksLength[]>([]);
    const [groupPromoCampaignsList, setPromoCampaignsList] = useState<BundleDtoWithLinksLength[]>([]);

    const checkedItems = useRef<Record<number, boolean>>({});
    const [selectedItems, setSelectedItems] = useState<CheckedItem[] | null>(null);
    const [selectedSection, setSelectedSection] = useState(BundleTypes.IDEA);

    const history = useHistory();

    const dataList = selectedSection === BundleTypes.IDEA ? groupBundlesList : groupPromoCampaignsList;
    const { current: copyDataList } = selectedSection === BundleTypes.IDEA ? copyBundleList : copyRelatedCampaigns;
    const setDataList = selectedSection === BundleTypes.IDEA ? setBundlesList : setPromoCampaignsList;

    const isSelect = selectedItems !== null;

    const toggleSelect = () => clearSelectedItems(isSelect);

    const loadData = async () => {
        setLoading(true);
        try {
            const { groups = [] } = await getCampaignGroupList();
            const { bundles, relatedCampaigns } = groups.reduce<BundlesRelatedCampaigns>((result, group) => {
                const key = group.type === BundleTypes.IDEA ? groupTypes.BUNDLE : groupTypes.RELATED_CAMPAIGN;
                return {
                    ...result,
                    [key]: [...result[key], { ...group, linksLength: group.links.length }],
                };
            }, { bundles: [], relatedCampaigns: [] });

            copyBundleList.current = bundles;
            copyRelatedCampaigns.current = relatedCampaigns;
            setBundlesList(bundles);
            setPromoCampaignsList(relatedCampaigns);
        } catch (err) {
            console.error(err);
        }
        setLoading(false);
    };

    useEffect(() => {
        loadData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const clearSelectedItems = (offSelect?: boolean, saveSelecting?: boolean) => {
        checkedItems.current = {};
        setSelectedItems(prev => offSelect || (saveSelecting && !prev) ? null : []);
    };

    const handleSelectItem = (selectItem: CheckedItem) => {
        setSelectedItems(prev => {
            if (!prev) {
                return prev;
            }

            const { id: selectedId, checked } = selectItem;
            checkedItems.current[selectedId] = checked;

            if (checked) {
                return [...prev, selectItem];
            }

            return prev.filter(({ id }) => id !== selectedId);
        });
    };

    const handleSelectAllItem = () => {
        const isAllChecked = dataList.length === selectedItems?.length;
        if (isAllChecked) {
            clearSelectedItems();
        } else {
            const newSelectedItems = dataList.map(({ id, name }) => {
                checkedItems.current[id] = true;
                return { id, name, checked: true };
            });
            setSelectedItems(newSelectedItems);
        }
    };

    const handleChangeSection = (e: RadioChangeEvent) => {
        const sectionType = e.target.value;
        clearSelectedItems(true);
        setSelectedSection(sectionType);
    };

    const refreshList = () => {
        loadData();
        clearSelectedItems(true);
    };

    const onAddGroupClick = () => {
        history.push(`${matchPath}${GROUPS_PAGES.ADD_GROUP}?type=${selectedSection}`);
    };

    const buttons: ButtonProps[] = [
        isSelect ? {
            type: 'primary',
            label: dataList.length && dataList.length === selectedItems?.length
                ? BUTTON_TEXT.CANCEL_ALL
                : BUTTON_TEXT.SELECT_ALL,
            onClick: handleSelectAllItem,
        } : {
            type: 'primary',
            label: BUTTON_TEXT.ADD,
            onClick: onAddGroupClick,
        },
        {
            label: isSelect ? BUTTON_TEXT.CANCEL : BUTTON_TEXT.SELECT,
            onClick: toggleSelect,
        },
    ];

    const onChangeInput = useCallback(() => {
        clearSelectedItems(false, true);
    }, []);

    return (
        <div className={cn(styles.container, { [styles.notEmpty]: copyDataList.length && !loading })}>
            <Header buttonBack={false} menuMode />
            <Radio.Group
                onChange={handleChangeSection}
                value={selectedSection}
                className={styles.section}
                options={GROUP_OPTIONS}
                optionType="button"
            />
            {loading ? (
                <div className={styles.loadingContainer}>
                    <div className={styles.loading}>
                        <SyncOutlined spin />
                    </div>
                </div>
            ) : !copyDataList.length ? (
                <EmptyGroup
                    onCreateClick={onAddGroupClick}
                    title={
                        selectedSection === BundleTypes.IDEA
                            ? TITLE.EMPTY_BUNDLES
                            : TITLE.EMPTY_PROMO_CAMPAIGNS
                    }
                    preTitle={
                        selectedSection === BundleTypes.IDEA
                            ? PRE_TITLE.CREATE_BUNDLE
                            : PRE_TITLE.CREATE_PROMO_CAMPAIGN
                    }
                />
            ) : (
                <>
                    <HeaderWithActions<BundleDtoWithLinksLength>
                        title={selectedSection === BundleTypes.IDEA ? TITLE.BUNDLE : TITLE.PROMO_CAMPAIGNS}
                        buttons={buttons}
                        triggerResetParams={selectedSection}
                        resetLabel={RESET_LABEL}
                        setDataList={setDataList}
                        copyDataList={copyDataList}
                        matchPath={matchPath}
                        inputPlaceholder={selectedSection === BundleTypes.IDEA ? SEARCH_INPUT.BUNDLE_SEARCH : SEARCH_INPUT.PROMO_SEARCH}
                        sortByFieldKey="name"
                        menuItems={DROPDOWN_SORT_MENU}
                        onChangeInput={onChangeInput}
                    />
                    <div className={styles.content}>
                        <GroupListByType
                            type={selectedSection}
                            checkedItems={checkedItems.current}
                            dataList={dataList}
                            matchPath={matchPath}
                            select={isSelect}
                            handleSelectItem={handleSelectItem}
                        />
                    </div>
                    {isSelect && (
                        <div className={styles.footer}>
                            <span>{`Выбрано групп: ${selectedItems!.length}`}</span>
                            <TableDeleteModal<CheckedItem>
                                sourceForRemove={selectedItems || []}
                                listIdForRemove={(selectedItems || []).map(item => item.id)}
                                deleteFunction={deleteCampaignGroup}
                                refreshTable={refreshList}
                                modalSuccessTitle={TITLE.MODAL_SUCCESS_TITLE}
                                modalTitle={TITLE.MODAL_TITLE}
                                listNameKey="name"
                            >
                                <Button
                                    type="primary"
                                    disabled={!selectedItems?.length}
                                    danger
                                >
                                    {BUTTON_TEXT.DELETE}
                                </Button>
                            </TableDeleteModal>
                        </div>
                    )}
                </>
            )}
        </div>
    );
};

export default GroupList;
