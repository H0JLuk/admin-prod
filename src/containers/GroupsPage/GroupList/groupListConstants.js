
export const TYPES = {
    IDEA: 'IDEA',
    ASSOCIATION: 'ASSOCIATION',
};

export const TITLE = {
    BUNDLE: 'Бандлы',
    PROMO_CAMPAIGNS: 'Связанные промо-кампании',
    MODAL_TITLE: 'Вы уверены что хотите удалить эти группы ?',
    MODAL_SUCCESS_TITLE: 'Результат удаления групп',
    EMPTY_BUNDLES: 'Нет созданных бандлов',
    EMPTY_PROMO_CAMPAIGNS: 'Нет созданных связанных промо-кампаний',
};

export const PRE_TITLE = {
    CREATE_BUNDLE: 'Создайте первый бандл прямо сейчас',
    CREATE_PROMO_CAMPAIGN: 'Создайте первую связанную промо-кампанию прямо сейчас',
};

export const BUTTON = {
    ADD: 'Добавить',
    CANCEL: 'Отменить',
    CREATE: 'Создать',
    CHOOSE: 'Выбрать',
    CHOOSE_ALL: 'Выбрать все',
    CHOOSE_ALL_REMOVE: 'Отменить выбор',
    DELETE: 'Удалить',
};

export const SEARCH_INPUT = {
    PROMO_SEARCH: 'Поиск промо-кампании',
    BUNDLE_SEARCH: 'Поиск бандла',
};

export const RESET_LABEL = 'По умолчанию';
export const DROPDOWN_SORT_MENU = [
    { name: 'name', label: 'По имени' },
    { name: 'linksLength', label: 'По кол-ву промо-кампаний' },
];

export const GROUP_OPTIONS = [
    { label: TITLE.BUNDLE, value: TYPES.IDEA },
    { label: TITLE.PROMO_CAMPAIGNS, value: TYPES.ASSOCIATION }
];

export const EMPTY_TABLE = {
    firstMessagePart: 'Мы ничего не нашли.',
    secondMessagePart: 'Измените значение поиска и попробуйте еще раз',
};

export const DIRECTION = {
    ASC: 'ASC',
    DESC: 'DESC',
};

export const sortByFieldKey = {
    NAME: 'name',
};

export const groupTypes = {
    BUNDLE: 'bundles',
    RELATED_CAMPAIGN: 'relatedCampaigns',
};