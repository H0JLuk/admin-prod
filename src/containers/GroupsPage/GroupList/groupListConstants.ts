import { BundleTypes } from '../groupPageConstants';

export const TITLE = {
    BUNDLE: 'Бандлы',
    PROMO_CAMPAIGNS: 'Связанные промо-кампании',
    MODAL_TITLE: 'Вы уверены что хотите удалить эти группы?',
    MODAL_SUCCESS_TITLE: 'Результат удаления групп',
    EMPTY_BUNDLES: 'Нет созданных бандлов',
    EMPTY_PROMO_CAMPAIGNS: 'Нет созданных связанных промо-кампаний',
};

export const PRE_TITLE = {
    CREATE_BUNDLE: 'Создайте первый бандл прямо сейчас',
    CREATE_PROMO_CAMPAIGN: 'Создайте первую связанную промо-кампанию прямо сейчас',
};

export const BUTTON = {
    ADD: 'Добавить',
    CANCEL: 'Отменить',
    CREATE: 'Создать',
    CHOOSE: 'Выбрать',
    CHOOSE_ALL: 'Выбрать все',
    CHOOSE_ALL_REMOVE: 'Отменить выбор',
    DELETE: 'Удалить',
};

export const SEARCH_INPUT = {
    PROMO_SEARCH: 'Поиск промо-кампании',
    BUNDLE_SEARCH: 'Поиск бандла',
};

export const RESET_LABEL = 'По умолчанию';
export const DROPDOWN_SORT_MENU = [
    { name: 'name', label: 'По имени' },
    { name: 'linksLength', label: 'По кол-ву промо-кампаний' },
];

export const GROUP_OPTIONS = [
    { label: TITLE.BUNDLE, value: BundleTypes.IDEA },
    { label: TITLE.PROMO_CAMPAIGNS, value: BundleTypes.ASSOCIATION },
];

export const EMPTY_TABLE = {
    firstMessagePart: 'Мы ничего не нашли.',
    secondMessagePart: 'Измените значение поиска и попробуйте еще раз',
};

export enum groupTypes {
    BUNDLE = 'bundles',
    RELATED_CAMPAIGN = 'relatedCampaigns',
}
