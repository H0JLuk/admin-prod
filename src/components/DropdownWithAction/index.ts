import DropdownWithAction from './DropdownWithAction';

export default DropdownWithAction;
