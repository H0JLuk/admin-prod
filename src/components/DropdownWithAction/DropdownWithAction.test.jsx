import React from 'react';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import { RESET_LABEL } from '../../containers/GroupsPage/GroupList/groupListConstants';
import DropdownWithAction from './DropdownWithAction';

const TEST_DROPDOWN_LABEL = 'Сортировать';

const TEST_MENU_ITEMS = [
    {
        name: 'TEST_NAME_ONE',
        label: 'TEST_LABEL_ONE',
        active: false
    },
    {
        name: 'TEST_NAME_TWO',
        label: 'TEST_LABEL_TWO',
        active: false
    }
];

const TEST_PROPS = {
    dropdownLabel: TEST_DROPDOWN_LABEL,
    onMenuItemClick: jest.fn(),
    menuItems: TEST_MENU_ITEMS,
    withReset: false,
    resetLabel: RESET_LABEL,
};

describe('test to open dropdown and select element', () => {
    const Dropdown = mount(<DropdownWithAction { ...TEST_PROPS } />);

    it('open dropdown', () => {
        const DropdownButton = Dropdown.find('button');
        DropdownButton.simulate('click');
        const OpenDropdown = Dropdown.find('button.ant-dropdown-open');
        expect(OpenDropdown.debug()).not.toBe('');
    });

    it('click menu item element', () => {
        const DropdownMenuItems = document.querySelectorAll('div .ant-dropdown .dropdownMenuItem');
        const DropdownMenuItem = document.querySelector('div .ant-dropdown .dropdownMenuItem');
        expect(DropdownMenuItem).toMatchSnapshot();
        expect(DropdownMenuItems).toHaveLength(2);

        act(() => {
            const overlay = Dropdown.childAt(0).prop('overlay');
            overlay.props.children[0][0].props.onClick();
            expect(TEST_PROPS.onMenuItemClick).toHaveBeenCalledWith(TEST_MENU_ITEMS[0].name);
            expect(TEST_PROPS.onMenuItemClick).toHaveBeenCalledTimes(1);
        });

    });
});
