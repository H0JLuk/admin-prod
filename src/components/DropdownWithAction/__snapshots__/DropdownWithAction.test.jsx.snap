// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to open dropdown and select element click menu item element 1`] = `
<div
  class="dropdownMenuItem"
>
  TEST_LABEL_ONE
</div>
`;
